### DISCUSSION TITLE
otterkit / GUI

### DISCUSSION THREAD

**Discussion by devjmfg** [otterkit/otterkit-cobol#30] [Feature Ideas]
The possibility of developing graphical interfaces is planned? 
Fujitsu's PowerCobol was the best approach I found to date in Cobol.
Best Regards,

**Comment by KTSnowy** [comment]
> The possibility of developing graphical interfaces is planned?

We're planning to eventually have a GUI library or framework as part of the class library that will be included with the compiler (Likely implemented as COBOL bindings for an existing native library).

This will be done as an added external library, instead of directly embedding it into the syntax or runtime. Due to the generally OS-specific nature of GUI code we feel that this would be the best choice, so we can keep Otterkit as platform independent as possible and also to be able to update the GUI code separately from both the compiler and runtime library.

Fujitsu's PowerCobol only worked natively on Windows if I recall correctly. We'll need a more portable solution that works for Windows, Linux and macOS.

Let me know if you have any recommendations for a native GUI library that we could use for this.

**Comment by devjmfg** [comment]
Cool :)
Regards and good job.
